CMakeLists.txt

```diff
@@ -512,6 +512,38 @@ if(ENABLE_HARDENING)
     if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
       try_append_linker_flag("-Wl,-fixup_chains" TARGET hardening_interface)
     endif()
+
+    if(HAVE_LIBCPP)
+      # https://libcxx.llvm.org/Hardening.html
+      if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+        target_compile_definitions(hardening_interface INTERFACE
+          _LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_DEBUG
+          _LIBCPP_ABI_BOUNDED_ITERATORS
+          _LIBCPP_ABI_BOUNDED_ITERATORS_IN_STRING
+          _LIBCPP_ABI_BOUNDED_ITERATORS_IN_VECTOR
+          _LIBCPP_ABI_BOUNDED_UNIQUE_PTR
+          _LIBCPP_ABI_BOUNDED_ITERATORS_IN_STD_ARRAY
+        )
+      else()
+        target_compile_definitions(hardening_interface INTERFACE _LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST)
```

Comment:
Seems like you copied your comment from https://github.com/bitcoin/bitcoin/pull/31424#discussion_r1871047821 to a new discussion thread here. I think that is confusing and will make it difficult to follow the conversation flow. I have already replied to your comment in the [original thread](https://github.com/bitcoin/bitcoin/pull/31424#discussion_r1875891496).

I am marking this as resolved. If you think the original thread was wrongly marked as resolved I think it would be better to comment there and I will reopen it (or feel free to reopen it yourself if github allows that).